src/bindings/typescript/serializers.ts

```diff
@@ -279,49 +279,63 @@ const addressSerializer: Serializer<{ optional: boolean }> = {
     },
 };
 
+function getCellLikeTsType(kind: "cell" | "slice" | "builder") {
+    return kind == "cell" ? "Cell" : kind == "slice" ? "Slice" : "Builder";
+}
+
 const cellSerializer: Serializer<{
     kind: "cell" | "slice" | "builder";
     optional: boolean;
 }> = {
     tsType(v) {
         if (v.optional) {
-            return "Cell | null";
+            return `${getCellLikeTsType(v.kind)} | null`;
         } else {
-            return "Cell";
+            return getCellLikeTsType(v.kind);
         }
     },
     tsLoad(v, slice, field, w) {
         if (v.optional) {
             w.append(
-                `let ${field} = ${slice}.loadBit() ? ${slice}.loadRef() : null;`,
+                `let ${field} = ${slice}.loadBit() ? ${slice}.loadRef()${v.kind !== "cell" ? ".as" + getCellLikeTsType(v.kind) + "()" : ""} : null;`,
```

Comment:
let's factor out this into a helper producing `asSlice`, `asBuilder`, ...
and also apply it for the cases below